[ENG-7204][docs] update CocoaPods Cache server documentation (#20923)
expo/expo: docs/pages/build-reference/caching.mdx

## iOS dependencies

> EAS Build by default cache **Podfile.lock** file to provide consistent results across managed app builds.

Reviewer: ```suggestion
EAS Build caches the **Podfile.lock** file by default. This provides consistent results across managed app builds.
```

Additionally EAS Build runs a CocoaPods cache server that can speed up downloading iOS dependencies for your build jobs. It also makes the service more resilient to CocoaPods CDN outages.

Cache server is configured to cache almost all the pods served from the official CocoaPods CDN, minus a few exceptions which can't be handled by our cache server. These exceptions can be found on the [blacklist](https://github.com/expo/eas-build/blob/main/packages/cocoapods-nexus-plugin/lib/cocoapods_plugin.rb#L4) of our custom CocoaPods plugin and are fetched directly from CDN instead of being fetched through our cache server.

> This is expermiental feature which requires explicit opt-in and can be **enabled** by setting `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"1"` in **eas.json**.

Reviewer: ```suggestion
This is expermiental feature which requires explicit opt-in. You can **enable** it by setting `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"1"` in **eas.json**.
```

{/* prettier-ignore */}
```json eas.json
{
  "build": {
    "production": {
      "env": {
        "EAS_BUILD_ENABLE_COCOAPODS_CACHE": "1"
        /* @hide ... */ /* @end */
      }
      /* @hide ... */ /* @end */
    }
    /* @hide ... */ /* @end */
  }
  /* @hide ... */ /* @end */
}
```

> To explicitly disable using our CocoaPods cache server for your builds set `EAS_BUILD_DISABLE_COCOAPODS_CACHE` env variable value to `"1"` in **eas.json**.

Reviewer: ```suggestion
To explicitly disable using our CocoaPods cache server for your builds set `EAS_BUILD_DISABLE_COCOAPODS_CACHE` env variable value to `"0"` in **eas.json**.
```
?

## iOS dependencies

EAS Build caches the **Podfile.lock** file by default. This provides consistent results across managed app builds.

> Additionally, EAS Build runs a CocoaPods cache server that can speed up downloading iOS dependencies for your build jobs. It also makes the service more resilient to CocoaPods CDN outages.

Reviewer 2: ```suggestion
Additionally, EAS Build runs a CocoaPods cache server that speeds up downloading iOS dependencies for your build jobs. It also makes the service more resilient to CocoaPods CDN outages.
```

Using the word "can" sounds like it may happen or not. If this something being run when creating a build, we can be more affirmative about that the caching will happen? Let me know if I'm understanding it in wrong way.

Author: In some cases using the CocoaPods cache server can speed up build time, but in some other cases, it might not happen. I think we are not sure that this will always be true, although it should happen in the majority of cases.

Reviewer 2: If that's the case, maybe we can rephrase it like this:

```
Additionally, EAS Build runs a CocoaPods cache server that can speed up downloading iOS dependencies for your build jobs in most cases.
```

Having that extra context also tells the user that it might not happen in their some cases.

The cache server is configured to cache almost all the pods served from the official CocoaPods CDN, minus a few exceptions that our cache server can't handle. These exceptions can be found on the [blacklist](https://github.com/expo/eas-build/blob/main/packages/cocoapods-nexus-plugin/lib/cocoapods_plugin.rb#L4) of our custom CocoaPods plugin. These are fetched directly from the CDN instead of being fetched through our cache server.

> This is expermiental feature which requires explicit opt-in. You can **enable** it by setting `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"1"` in **eas.json**.

Reviewer 2: ```suggestion
This is an experimental feature that requires explicit opt-in. You can **enable** it by setting the `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"1"` in **eas.json**.
```

> This is an expermiental feature that requires explicit opt-in. You can **enable** it by setting the `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"1"` in **eas.json**.

Reviewer 2: ```suggestion
This is an experimental feature that requires explicit opt-in. You can **enable** it by setting the `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"1"` in **eas.json**.
```
I think this typo wasn't resolved in one of the previous commits 😄

{/* prettier-ignore */}
```json eas.json
{
  "build": {
    "production": {
      "env": {
        "EAS_BUILD_ENABLE_COCOAPODS_CACHE": "1"
        /* @hide ... */ /* @end */
      }
      /* @hide ... */ /* @end */
    }
    /* @hide ... */ /* @end */
  }
  /* @hide ... */ /* @end */
}
```

> To explicitly disable using our CocoaPods cache server for your builds set `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"0"` in **eas.json**.

Reviewer 2: ```suggestion
To explicitly disable using our CocoaPods cache server for your builds, set the `EAS_BUILD_ENABLE_COCOAPODS_CACHE` env variable value to `"0"` in **eas.json**.
```
